data-prepper-plugins/mongodb/src/main/java/org/opensearch/dataprepper/plugins/mongo/configuration/MongoDBSourceConfig.java

```diff
@@ -0,0 +1,109 @@
+package org.opensearch.dataprepper.plugins.mongo.configuration;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import jakarta.validation.constraints.NotNull;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class MongoDBSourceConfig {
+    private static final int DEFAULT_PORT = 27017;
+    private static final Boolean SSL_ENABLED = false;
+    private static final Boolean SSL_INVALID_HOST_ALLOWED = false;
+    private static final String DEFAULT_SNAPSHOT_FETCH_SIZE = "1000";
+    private static final String DEFAULT_READ_PREFERENCE = "secondaryPreferred";
+    @JsonProperty("hostname")
+    private @NotNull String hostname;
+    @JsonProperty("port")
+    private int port = DEFAULT_PORT;
+    @JsonProperty("trust_store_file_path")
+    private String trustStoreFilePath;
+    @JsonProperty("trust_store_password")
+    private String trustStorePassword;
+    @JsonProperty("credentials")
+    private CredentialsConfig credentialsConfig;
+
+    @JsonProperty("snapshot_fetch_size")
+    private String snapshotFetchSize;
+    @JsonProperty("read_preference")
+    private String readPreference;
+    @JsonProperty("collections")
+    private List<CollectionConfig> collections;
+    @JsonProperty("acknowledgments")
+    private Boolean acknowledgments = false;
+    @JsonProperty("ssl")
+    private Boolean ssl;
+    @JsonProperty("ssl_invalid_host_allowed")
```

Comment:
In peer-forwarder, we have a similar configuration named: `ssl_insecure_disable_verification`. Maybe we should use that same name.

Reply:
Done